pkg/config/example_test.go

```diff
@@ -0,0 +1,265 @@
+	data, err := proto.Marshal(pb)
+	if err != nil {
+		panic(err)
+	}
+
+	return data
+}
+
+// createSigningIdentity returns a identity that can be used for signing transactions.
+// Signing identity can be retrieved from MSP configuration for each peer.
+func createSigningIdentity() *config.SigningIdentity {
+	privKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to generate private key: %v", err))
+	}
+
+	return &config.SigningIdentity{
+		Certificate: generateCert(),
+		PrivateKey:  privKey,
+		MSPID:       "ID",
+	}
+}
+
+// Generate a certificate for the SigningIdentity.
+func generateCert() *x509.Certificate {
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+
+	if err != nil {
+		log.Fatalf("Failed to generate serial number: %s", err)
+	}
+
+	return &x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			CommonName:   "Wile E. Coyote",
+			Organization: []string{"Acme Co"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(365 * 24 * time.Hour),
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+}
+
+func Example_usage() {
+	// Retrieve the config for the channel
+	baseConfig := fetchChannelConfig()
+	// The config that will be updated
+	updatedConfig := proto.Clone(baseConfig).(*cb.Config)
+
+	// Must retrieve the current orderer configuration from block and modify
+	// the desired values
+	orderer := &config.Orderer{
+		OrdererType: config.ConsensusTypeKafka,
+		Kafka: config.Kafka{
+			Brokers: []string{"kafka0:9092", "kafka1:9092"},
+		},
+		Organizations: []*config.Organization{
+			{
+				Name: "OrdererOrg",
+				Policies: map[string]*config.Policy{
+					config.AdminsPolicyKey: {
+						Type: config.ImplicitMetaPolicyType,
+						Rule: "Majority Endorsement",
+					},
+					config.ReadersPolicyKey: { // Adding Readers policy
+						Type: config.ImplicitMetaPolicyType,
+						Rule: "Majority Endorsement",
+					},
+				},
+			},
+		},
+		BatchSize: config.BatchSize{
+			MaxMessageCount:   500, // Updating from 100
+			AbsoluteMaxBytes:  100,
+			PreferredMaxBytes: 100,
+		},
+		Addresses: []string{"127.0.0.1:7050"},
+	}
+
+	err := config.UpdateOrdererConfiguration(updatedConfig, orderer)
+	if err != nil {
+		panic(nil)
+	}
+
+	newAnchorPeer := &config.AnchorPeer{
+		Host: "127.0.0.2",
+		Port: 7051,
+	}
+
+	//	Add a new anchor peer
+	err = config.AddAnchorPeer(updatedConfig, "Org1MSP", newAnchorPeer)
+	if err != nil {
+		panic(err)
+	}
+
+	// Remove an old anchor peer from Org1MSP
+	oldAnchorPeer := &config.AnchorPeer{
+		Host: "127.0.0.2",
+		Port: 7051,
+	}
+
+	err = config.RemoveAnchorPeer(updatedConfig, "Org1MSP", oldAnchorPeer)
+	if err != nil {
+		panic(err)
+	}
+
```

Comment:
At some point we should also add examples for updating the MSPs (adding cas/revoking certs etc). I believe that story was done in parallel with this and there was no acceptance criteria for adding an example.